src/TracingManualInstrumentation/TracingManualInstrumentation.cs

```diff
@@ -0,0 +1,374 @@
+                CountdownEvent.Wait();
+            }
+        }
+
+        /// <summary>
+        /// This event handler is invoked by Solace Systems Messaging API when a message arrives
+        /// </summary>
+        /// <param name="source"></param>
+        /// <param name="args"></param>
+        private void HandleMessageEvent(object source, MessageEventArgs args)
+        {
+            // Received a message
+            Console.WriteLine("Received message.");
+            using (IMessage message = args.Message)
+            {
+                // Expecting the message content as a binary attachment
+                Console.WriteLine("Message content: {0}", Encoding.ASCII.GetString(message.BinaryAttachment));
+
+                //Extracts tracing context from a message, if any using the SolaceMessageCarrier.Getter. 
+                // propagationContext comprises of context as well as baggage extracted from the message
+                PropagationContext propagationContext = compositePropagator.Extract(default, message, SolaceMessageCarrier.Getter);
+
+                HowToCreateActivityOnMessageReceive(message, propagationContext);
+                // ACK the message
+                Flow.Ack(message.ADMessageId);
+                CountdownEvent.Signal();
+            }
+        }
+
+        /// <summary>
+        /// This event handler is invoked by Solace Systems Messaging API when a flwo event happens
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="args"></param>
+        public void HandleFlowEvent(object sender, FlowEventArgs args)
+        {
+            // Received a flow event
+            Console.WriteLine("Received Flow Event '{0}' Type: '{1}' Text: '{2}'",
+                args.Event,
+                args.ResponseCode.ToString(),
+                args.Info);
+        }
+
+        #region IDisposable Support
+        private bool disposedValue = false; // To detect redundant calls
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    Flow.Dispose();
+                }
+                disposedValue = true;
+            }
+        }
+        public void Dispose()
+        {
+            Dispose(true);
+        }
+
+        Activity HowToCreateActivityOnMessagePublish(IMessage message)
+        {
+            // .NET OpenTelemetry API rely on System.Diagnostics.Activity and System.Diagnostics.ActivityContext for creation of activity and context
+            var activityName = "ProducerActivity";
+            using var activity = activitySource.StartActivity(activityName, ActivityKind.Producer);
+            activity?.SetTag("messaging.system", "Solace Pubsub plus");
+            activity?.SetTag("messaging.destination_kind", "queue");
+            activity?.SetTag("messaging.destination", "queueName");
```

Comment:
Can we also make sure we use `solace` for the messaging.system since that is what we already have in the JCSMP samples. 

@TamimiGitHub - FYI here. we will probably want to double check other samples and make sure we follow this same example whenever we create content